Assets/Scripts/Utility/AssetInjection/WorldDataReplacement.cs

```diff
@@ -485,6 +486,23 @@ public static int MakeLocationKey(int regionIndex, int locationIndex)
             return (locationIndex * 100) + regionIndex;
         }
 
+        /// <summary>
+        /// Applies automap data from that defined in building replacement data. Values of '30' will be ignored allowing only the relevant parts to be modified.
+        /// </summary>
+        /// <param name="buildingData">BuildingReplacementData input</param>
+        /// <param name="blockAutoMapData">A reference to the block automap data byte array to merge into</param>
+        public static void ApplyBuildingReplacementAutoMapData(BuildingReplacementData buildingData, ref byte[] blockAutoMapData)
+        {
+            if (buildingData.AutoMapData != null && buildingData.AutoMapData.Length == AutoMapDataSize)
+            {
+                for (int i = 0; i < AutoMapDataSize; i++)
+                {
+                    if (buildingData.AutoMapData[i] != 30)
```

Comment:
I wish we had an enum for these values, but I think I can see from AutomapEditor.cs `GetColor` what values are used for what. 30 does seem out of supported range